core/src/main/java/bio/terra/pearl/core/service/survey/SurveyTaskDispatcher.java

```diff
@@ -131,4 +135,43 @@ protected Optional<SurveyTaskConfigDto> findTaskConfigByStableId(UUID studyEnvir
 
         return Optional.of(new SurveyTaskConfigDto(sesOpt.get(), surveyOpt.get()));
     }
+
+    @Override
+    public void copyTaskData(ParticipantTask newTask, ParticipantTask oldTask, SurveyTaskConfigDto taskDispatchConfig) {
+        super.copyTaskData(newTask, oldTask, taskDispatchConfig);
+
+        newTask.setSurveyResponseId(null);
+        newTask.setCompletedAt(null);
+
+        // if the survey is set to prepopulate, copy the answers from the old task to the new task
+        // we need to also create a new survey response for the new task and attach it to that task
+        if(taskDispatchConfig.getSurvey().isPrepopulate()) {
+            Optional<SurveyResponse> priorResponse = surveyResponseService.findOneWithAnswers(oldTask.getSurveyResponseId());
+
+            if (priorResponse.isPresent() && taskDispatchConfig.getSurvey().isPrepopulate()) {
+                SurveyResponse createdResponse = createPrepopulatedSurveyResponse(priorResponse.get());
+                newTask.setStatus(TaskStatus.NEW);
+                newTask.setSurveyResponseId(createdResponse.getId());
+            }
+        }
+    }
```

Comment:
I feel like this would be useful to have a test in place, I could see this code getting moved/forgotten